sim/side.ts

```diff
@@ -629,17 +629,24 @@ export class Side {
 
 			switch (choiceType) {
 			case 'move':
-				let targetLoc = 0;
-				if (/\s-?[1-3]$/.test(data)) {
-					targetLoc = parseInt(data.slice(-2), 10);
-					data = data.slice(0, data.lastIndexOf(' '));
+				let [targetLoc, willMega, willUltra, willZ] = [0, '', '', ''];
```

Comment:
I still don't like this initialization.

```suggestion
				let targetLoc = 0;
				let megaOrZ = '';
```

Reply:
I actually explicitly think `willMega, willUltra, willZ` should be combined into one variable – it's done when `chooseMove` is called, anyway.